Extension/lib/content-script/preload.js

```diff
@@ -480,15 +691,13 @@
         }
         delete collapseRequests[response.requestId];
 
-        if (response.collapse !== true) {
-            // Return element visibility in case if it should not be collapsed
-            toggleElement(collapseRequest.element);
-            return;
+        if (response.collapse === true) {
+            var element = collapseRequest.element;
+            elementCollapser.collapseElement(element);
         }
 
-        var element = collapseRequest.element;
-        var tagName = collapseRequest.tagName;
-        collapseElement(element, tagName);
+        // In any case we should remove hiding style
+        elementCollapser.unhideElement(collapseRequest.element);

```

Comment:
Key points:
1. In the end I'd like to have our "style" removed from the page code
2. Collapse is "tougher" than hiding as hidden elements DOM path may change in future so it won't work after that.